src/features/registry.js

```diff
@@ -12,6 +14,11 @@ systemJSPrototype.get = function (id) {
 };
 
 systemJSPrototype.set = function (id, module) {
+  try {
+    new URL(id, baseUrl);
+  } catch (err) {
+    console.warn(Error(process.env.SYSTEM_PRODUCTION ? errMsg(10, id) : errMsg(10, 'Invalid module id - ' + id + ' must be url')));
+  }
```

Comment:
We discussed this whole error should be development-mode - eg `if (!process.env.SYSTEM_PRODUCTION)` around the whole check.

My main concern with making it on by default is if users want to use non-URL registry entries they are welcome to... there is nothing in the ECMA262 specification that says specifiers have to be URLs. It is simply the convention in the browser.

Reply:
👍 updated